app/models/non_hmis_assessment.rb

```diff
@@ -23,6 +23,18 @@ class NonHmisAssessment < ActiveRecord::Base
     where(type: covid_assessment_types)
   end
 
+  scope :visible_by, ->(user) do
+    where(non_hmis_client_id: NonHmisClient.visible_to(user).select(:id))
+  end
+
+  scope :editable_by, ->(user) do
+    where(non_hmis_client: NonHmisClient.editable_by(user).select(:id))
+  end
```

Comment:
This needs to inspect the assessment to determine if it was created by someone at your agency.  If so, then you can edit.